QuestionAnswering/QA_ICL/backend/fastapi_app/controllers/chat.py

```diff
@@ -20,10 +20,15 @@
 
 def binary_search(low: int, high: int, fn: Callable[[int], int], target: int):
     """Performs binary search over the integer space.
+
     Args:
-        low: lower bound of the search space
-        high: upper bound of the search space
-        fn: a monotonic function over integer domain and range
+        low (int): Lower bound of the search space.
+        high (int): Upper bound of the search space.
+        fn (Callable[[int], int]): Monotonic function over integer domain and range.
+        target (int): Target value to search for.
```

Comment:
Is there a certain style guide that you wish to follow for adding docstring? E.g. per [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html?showone=Imports_formatting#doc-function-args), "The description should include required type(s) if the code does not contain a corresponding type annotation." 
Generally speaking, we should only declare the types once either in the docstring or the type-hints. The only case where it would benefit from declaring twice is when we want to benefit from static type checks of type-hints and document a public API. However, in our case, the software is used via HTTP endpoints, not as a Python package. For that reason, I would advocate for only declaring types in the type-hints only to benefit the developer's experience, not the user's experience; the type-hints will show up in IDEs that support Python/mypy.